core/common/src/test/java/alluxio/collections/IndexedSetTest.java

```diff
@@ -249,13 +249,43 @@ public void addTheSameObjectMultipleTimesTest() {
    */
   @Test
   public void iteratorRemoveTest() {
-    Iterator<Pair> it =  mSet.iterator();
+    long removed = 0;
+    Iterator<Pair> it = mSet.iterator();
     Assert.assertTrue(it.hasNext());
     final Pair first = it.next();
-    Set<Pair> allWithSameIntValue = mSet.getByField(mNonUniqueIntIndex, first.intValue());
-    Assert.assertTrue("Element should be in the set", allWithSameIntValue.contains(first));
+
+    Set<Pair> valueSet = mSet.getByField(mNonUniqueIntIndex, first.intValue());
+    Assert.assertTrue("Element should be in the set", valueSet.contains(first));
+
+    valueSet = mSet.getByField(mUniqueLongIndex, first.longValue());
+    Assert.assertTrue("Element should be in the set", valueSet.contains(first));
+
     it.remove();
-    allWithSameIntValue = mSet.getByField(mNonUniqueIntIndex, first.intValue());
-    Assert.assertFalse("Element should not be in the set", allWithSameIntValue.contains(first));
+    removed++;
+    valueSet = mSet.getByField(mNonUniqueIntIndex, first.intValue());
+    Assert.assertFalse("Element should not be in the set", valueSet.contains(first));
+
+    valueSet = mSet.getByField(mUniqueLongIndex, first.longValue());
+    Assert.assertFalse("Element should not be in the set", valueSet.contains(first));
+
+    while (it.hasNext()) {
+      it.next();
+      it.remove();
+      removed++;
+    }
+    Assert.assertEquals(removed, 9L);
+    Assert.assertTrue(mSet.size() == 0);
+
+    for (Pair o : mSet) {
```

Comment:
from here, we are testing foreach rather than testing remove.
It should be an individual test case similar as `iteratorRemoveTest`, maybe `iteratorForeachTest`